### ISSUE TITLE
GNS3 (2.13 initially OK on ubuntu 16.04) does not start after an upgrade to ubuntu 18.04 

### ISSUE THREAD

**Issue by e-thibault** [GNS3/gns3-gui#3274]
I am opening this new issue to the suggestion of Joe Higgins GNS3 Moderator in that same issue I opened in gns3.com/community
at the url:
https://gns3.com/community/featured/gns3-2-13-initialy-on-ubuntu-16-04-does-not-work-after-upgrade-to-ubuntu-18-04

initial situation:
ubuntu 16.04 / GNS3 2.13 OK

I wanted to make tests with FRR in a Docker container but I saw docker does not support anymore ubuntu 16.04 so I upgraded in
ubuntu 18.04.
After that GNS3 could not start so I performed an installation of GNS3, but then GNS3 would not start either.
I found a post in the forum with a similar issue:
https://gns3.com/community/featured/gns3-doesnt-work-after-updating-
so I tried a solution described in  the comments by Afaq Babar Apr 26 2016 at 10:24pm
i.e:
sudo apt-get purge python-pip
sudo apt-get clean
sudo apt-get autoclean
sudo apt-get remove
sudo apt-get autoremove
sudo apt-get update
sudo apt-get upgrade

which resulted in an upgrade of GNS3 in 2.2.28 but with the same issue.

Now the situation is:
ubuntu 18.04
Running GNS3 version 2.2.28 on Linux (64-bit) with Python 3.6.9 Qt 5.9.5 and PyQt 5.10.1.
$ apt list --installed | grep gns3

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

gns3-gui/bionic,now 2.2.28~bionic1 amd64  [installé]
gns3-iou/bionic,now 0.0.3~bionic1 amd64  [installé]
gns3-server/bionic,now 2.2.28~bionic1 amd64  [installé]

You can find more details at:
https://gns3.com/community/featured/gns3-2-13-initialy-on-ubuntu-16-04-does-not-work-after-upgrade-to-ubuntu-18-04

but I launched gns3 in debug mode and here is the result:
$ gns3 --debug
2021-12-15 19:33:59 INFO root:126 Log level: DEBUG
2021-12-15 19:33:59 INFO main:259 GNS3 GUI version 2.2.28
2021-12-15 19:33:59 INFO main:260 Copyright (c) 2007-2021 GNS3 Technologies Inc.
2021-12-15 19:33:59 INFO main:261 Application started with /usr/bin/gns3 --debug
2021-12-15 19:33:59 DEBUG utils.analytics:123 Send stats to Google Analytics: v=1&tid=UA-55817127-3&cid=e87e71be-efe1-43a4-9aa7-b81bba2ffeb2&aip=1&t=screenview&an=GNS3&av=2.2.28&ua=Mozilla/5.0%20%28X11%3B%20Linux%20x86_64%29%20AppleWebKit/537.36%20%28KHTML%2C%20like%20Gecko%29%20Safari/537.36%20%20GNS3/2.2.28&cd=Main%20Window&ds=gns3-gui&sr=1920x1200&ul=fr_fr
2021-12-15 19:33:59 DEBUG utils.analytics:123 Send stats to Google Analytics: v=1&tid=UA-55817127-3&cid=e87e71be-efe1-43a4-9aa7-b81bba2ffeb2&aip=1&t=screenview&an=GNS3&av=2.2.28&ua=Mozilla/5.0%20%28X11%3B%20Linux%20x86_64%29%20AppleWebKit/537.36%20%28KHTML%2C%20like%20Gecko%29%20Safari/537.36%20%20GNS3/2.2.28&cd=Main%20Window&ds=gns3-gui&sr=1920x1200&ul=fr_fr
2021-12-15 19:33:59 DEBUG http_client:761 GET http://localhost:3080/v2/version
2021-12-15 19:33:59 DEBUG http_client:788 Error while connecting to local server Connection refused
2021-12-15 19:33:59 DEBUG http_client:761 GET http://localhost:3080/v2/version
2021-12-15 19:33:59 DEBUG http_client:788 Error while connecting to local server Connection refused
2021-12-15 19:33:59 DEBUG http_client:761 GET http://localhost:3080/v2/version
2021-12-15 19:33:59 DEBUG http_client:788 Error while connecting to local server Connection refused
2021-12-15 19:33:59 DEBUG local_server:477 Starting local server process with "/usr/bin/gns3" --local --debug --log="/home/thibault/.config/GNS3/2.2/gns3_server.log" --pid="/home/thibault/.config/GNS3/2.2/gns3_server.pid"
2021-12-15 19:33:59 DEBUG local_server:490 Local server process has started (PID=3675)
2021-12-15 19:33:59 DEBUG utils.progress_dialog:71 Set worker inside a thread <class 'gns3.utils.wait_for_connection_worker.WaitForConnectionWorker'>
2021-12-15 19:33:59 DEBUG utils.progress_dialog:112 WaitForConnectionWorker thread started
2021-12-15 19:34:04 ERROR local_server:496 Local server process has stopped
2021-12-15 19:34:04 ERROR local_server:498 usage: gns3 [-h] [--version] [--debug] [-q] [--config CONFIG]
            [--profile PROFILE]
            [path]
gns3: error: unrecognized arguments: --local --log=/home/thibault/.config/GNS3/2.2/gns3_server.log --pid=/home/thibault/.config/GNS3/2.2/gns3_server.pid

2021-12-15 19:34:20 DEBUG utils.progress_dialog:124 WaitForConnectionWorker thread canceled
2021-12-15 19:34:20 DEBUG utils.progress_dialog:149 WaitForConnectionWorker thread is being destroyed
2021-12-15 19:34:20 DEBUG utils.progress_dialog:156 WaitForConnectionWorker thread destroyed
2021-12-15 19:34:20 DEBUG utils.analytics:123 Send stats to Google Analytics: v=1&tid=UA-55817127-3&cid=e87e71be-efe1-43a4-9aa7-b81bba2ffeb2&aip=1&t=screenview&an=GNS3&av=2.2.28&ua=Mozilla/5.0%20%28X11%3B%20Linux%20x86_64%29%20AppleWebKit/537.36%20%28KHTML%2C%20like%20Gecko%29%20Safari/537.36%20%20GNS3/2.2.28&cd=Project%20Dialog&ds=gns3-gui&sr=1920x1200&ul=fr_fr
2021-12-15 19:34:30 DEBUG main_window:1125 Close the Main Window
2021-12-15 19:34:30 DEBUG utils.analytics:123 Send stats to Google Analytics: v=1&tid=UA-55817127-3&cid=e87e71be-efe1-43a4-9aa7-b81bba2ffeb2&aip=1&t=screenview&an=GNS3&av=2.2.28&ua=Mozilla/5.0%20%28X11%3B%20Linux%20x86_64%29%20AppleWebKit/537.36%20%28KHTML%2C%20like%20Gecko%29%20Safari/537.36%20%20GNS3/2.2.28&cd=Main%20Window&ds=gns3-gui&sc=end&sr=1920x1200&ul=fr_fr
2021-12-15 19:34:30 DEBUG main_window:1140 _finish_application_closing
2021-12-15 19:34:30 DEBUG local_config:387 Section MainWindow has not changed. Skip saving configuration
2021-12-15 19:34:30 DEBUG main_window:1151 Time spend in the software is 31

There are several issues:
- port 3080 of gns3 server
- files that are not recognized arguments

The latter being probably caused somehow (?) by the fact I run a fresh installation commands on a system where
GNS3 had already been  installed though not working anymore.

There are some more details in the gns3.com/community post but feel free to ask me more details to help solve this please

**Comment by e-thibault** [comment]
OK For troubleshooting purposes and comparison with my install on ubuntu18.04 at home,
a colleague set me a brand new VM with low ressources with ubuntu18.04.

I followed the instructions for a fresh installation below just like I did on my ubuntu 18.04:
(the difference with my PC is that is had already GNS3 installed)

sudo add-apt-repository ppa:gns3/ppa
sudo apt-get update
sudo apt-get install gns3-gui

and as I suspected the installation works fine on a fresh ubuntu 18.04
I have GNS 2.2.28 installed
with Python 3.6.9 Qt 5.9.5 and PyQt5.10.1.

=> so we can definitely rule out an intrisic issue with GNS2.2.28
a fresh install on a fresh ubunt 18.04 works so the issue here was caused by the fact that 
I ran the GNS3 commands for a fresh install on a PC that had already GNS3 installed
(that used to work with ubuntu 16.04), but which is broken now after the upgrade in 18.04.

So what is the best way forward now ?

**Comment by grossmj** [comment]
Backup the config files in `~/.config/GNS3/2.2` and try to install from scratch again to see if the problem is solved.

**Comment by e-thibault** [comment]
Thanks for your reply and thanks for all what you do for GNS3 Jeremy

So to backup the files and emulate a deletion of the directory, I renammed directory ~/.config/GNS3 to ~/.config/gons3

N.B. when I installed GNS3 2.2 the first time I have remarked that files in ~/.config/GNS3/2.2 were in fact inspired from the file with the same name in  ~/.config/GNS3 namely gns3_gui.conf
so in order to make the reinstallation of GNS3 look more like a fresh installation I thought it was better to suppress all traces of GNS3 in the directory  ~/.config/ hence renamming the directory GNS3 into gons3 (o for old).
I thought GNS install would not see that directory and do as if installation was brand new from configuration standpoint.

so with that performed, I installed gns3 with:
sudo add-apt-repository ppa:gns3/ppa
sudo apt-get update
sudo apt-get install gns3-gui

that last command resulted in:
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances       
Lecture des informations d'état... Fait
gns3-gui est déjà la version la plus récente (2.2.28~bionic1).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 0 non mis à jour.

Then when I launched gns3 I got:

thibault@R2D2compact:~$ gns3
2021-12-28 16:13:46 INFO root:126 Log level: INFO
2021-12-28 16:13:46 INFO main:259 GNS3 GUI version 2.2.28
2021-12-28 16:13:46 INFO main:260 Copyright (c) 2007-2021 GNS3 Technologies Inc.
2021-12-28 16:13:46 INFO main:261 Application started with /usr/bin/gns3
2021-12-28 16:14:18 CRITICAL utils.progress_dialog:181 WaitForConnectionWorker thread stopping with an error: Could not connect to localhost on port 3080: [Errno 111] Connexion refusée
2021-12-28 16:14:19 CRITICAL progress_dialog.py:183 Could not connect to localhost on port 3080: [Errno 111] Connexion refusée

It was impossible to connect to port 3080 so I quit GNS3.

so what, could be the next step ?
uninstall gns3 then reinstall, but I am not knwoledgeable very much in linux I don't know the commands to uninstall gns3 to be able to reinstall with a fresh install afterwards

Best Regards

Thibault

**Comment by e-thibault** [comment]
Hello,
**Happy new year 2022 !**

I googled to search for unisntall GNS3 and I stumbled upon this very good post:
https://www.thelinuxfaq.com/ubuntu/ubuntu-17-04-zesty-zapus/gns3?type=uninstall

In my case I just performed the command:
sudo apt-get remove gns3

then I reinstalled fully gns3 with the commands:

sudo add-apt-repository ppa:gns3/ppa
sudo apt-get update
sudo apt-get install gns3-gui

chmod 755 /usr/bin/dumpcap

and **now it works !**
even my past projects (with dynamips and IOU works)
thanks every one who helped

**N.B. I think it would be very important to make somewhere a document or a thread about upgrades
I. upgrade of GNS3, what should we do to keep our projects working (depending dynamips, IOU, etc.)
II. upgrade of ubuntu (i.e next version) what should we do to prepare GNS3 so that when we upgrade ubuntu we keep our projects working !**

I must say every time I upgraded I took quite some time to get back the projects working, best regards
Thibault
